tests/mdev_configuration_allocation_test.go

```diff
@@ -352,6 +356,173 @@ var _ = Describe("[Serial][sig-compute]MediatedDevices", Serial, decorators.VGPU
 			Eventually(checkAllMDEVCreated(newDesiredMdevTypeName, newExpectedInstancesNum), 3*time.Minute, 15*time.Second).Should(BeInPhase(k8sv1.PodSucceeded))
 
 		})
+		It("should successfully create VMI when guest memory is less than or equal to request memory", func() {
+			By("Creating a Fedora VMI")
+			vmi = tests.NewRandomFedoraVMI()
+			vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse("1G")
+			vGPUs := []v1.GPU{
+				{
+					Name:       "gpu1",
+					DeviceName: deviceName,
+				},
+			}
+			vmi.Spec.Domain.Devices.GPUs = vGPUs
+
+			guestMemory := resource.MustParse("512M")
+			vmi.Spec.Domain.Memory = &v1.Memory{Guest: &guestMemory}
+			vmi.Spec.Domain.Resources = v1.ResourceRequirements{}
```

Comment:
Try to avoid the usage of deprecated NewRandomFedoraVMI().
We can do something like this:
```suggestion
			networkData := libnet.CreateDefaultCloudInitNetworkData()
			vGPU := v1.GPU{
				Name:       "gpu1",
				DeviceName: deviceName,
			}
			vmi = libvmi.New(
				libvmi.WithRng(),
				libvmi.WithContainerDisk("disk0", cd.ContainerDiskFor(cd.ContainerDiskFedoraTestTooling)),
				libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
				libvmi.WithNetwork(v1.DefaultPodNetwork()),
				libvmi.WithCloudInitNoCloudNetworkData(networkData),
				libvmi.WithGuestMemory("512M"),
				withGPUDevice(vGPU),
			)
```
adding a new Option in this package:
```
// withGPUDevice adds gpu device.
func withGPUDevice(vGPU v1.GPU) libvmi.Option {
	return func(vmi *v1.VirtualMachineInstance) {
		vmi.Spec.Domain.Devices.GPUs = append(vmi.Spec.Domain.Devices.GPUs, vGPU)
	}
}
```

Reply:
Done